pkg/correlation/interceptor.go

```diff
@@ -0,0 +1,25 @@
+package correlation
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+// ContextInterceptor represents a correlation interceptor
+type ContextInterceptor struct {
+	Origin Component
+}
+
+// UnaryInterceptor creates a gRPC interceptor for adding
+// correlation ID to each request
+func (ci *ContextInterceptor) ContextUnaryInterceptor(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	ctx = NewContext(ctx, ci.Origin)
```

Comment:
Should there should be a bug-check here if Origin is not set or something.

Reply:
I think we're fine without that for now to keep the API clean. I don't want to fail a request because our logging is having issues.